lib/services/api/user_token_service.rb

```diff
@@ -23,9 +23,13 @@ def api_config
       @api_config ||= ::Settings[base_config[:module]].to_hash
     end
 
-    def generate_token(userid, requester_type, token_ttl: nil)
-      userid = userid.downcase
-      validate_userid(userid)
+    def generate_token(user_or_id, requester_type, token_ttl: nil)
+      if user_or_id.kind_of?(User)
+        userid = user_or_id.userid.downcase
```

Comment:
do we still need to do the validation? (at least the "in my region" aspect of what the validate method does)?

Reply:
I think I may have screwed up the `if` here, and the `end` should be a line earlier... let me look into this...

(sorry I took so long to respond to this...)